src/core/WeakMap.ts

```diff
@@ -0,0 +1,38 @@
+let wmUniqueIndex = 0;
+
+export default class WeakMap<K extends object, V> {
+
+    protected _id: string;
+
+    constructor() {
+        this._id = '__\0ec_wm_' + wmUniqueIndex++ + '_' + Math.random().toFixed(5);
+    }
```

Comment:
Id generating strategy can be same to https://github.com/apache/incubator-echarts/blob/next/src/util/model.ts#L721